lib/wasi/src/state/mod.rs

```diff
@@ -491,6 +492,15 @@ impl WasiFs {
         }
     }
 
+    /// Internal part of the core path resolution function which implements path
+    /// traversal logic such as going resolving relative path segments (such as
+    /// `.` and `..`) and resolving symlinks (while preventing infinite
+    /// loops/stack overflows).
+    ///
```

Comment:
I think something this complicated should probably have a "given X produces Y" description, like, given a path and an open file descriptor of the base directory, returns the inode of the file that it refers to, or an error if it doesn't exist or an error occurred (such as infinite recursion resolving symlinks).